Feature/telemetry data docs

roblim opened the pull request:
Addresses: https://app.asana.com/0/1119694864223484/1166012364881928/f

roblim committed:
Add filter to add date and data_context_id to url

roblim committed:
Pull data_context_id from data_context

- add as attribute to SiteBuilder
- pass to DefaultSiteSectionBuilder and DefaultSiteIndexBuilder in __init__ and add as attribute

roblim committed:
Pass data_context_id as arg when calling view.render

roblim committed:
Change to ‘jinja_context’ to be more clear

roblim committed:
Add tests for data docs telemetry

roblim committed:
Set anonymized_usage_data key in config

pep8speaks commented:
Thank you for submitting to Great Expectations!



* In the file [`great_expectations/core/logging/usage_statistics.py`](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/great_expectations/core/logging/usage_statistics.py):

> [Line 15:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/great_expectations/core/logging/usage_statistics.py#L15): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (131 > 120 characters)

* In the file [`great_expectations/data_context/data_context.py`](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/great_expectations/data_context/data_context.py):

> [Line 134:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/great_expectations/data_context/data_context.py#L134): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (125 > 120 characters)
> [Line 151:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/great_expectations/data_context/data_context.py#L151): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (145 > 120 characters)





* In the file [`great_expectations/render/renderer/site_builder.py`](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/great_expectations/render/renderer/site_builder.py):

> [Line 104:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/great_expectations/render/renderer/site_builder.py#L104): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (129 > 120 characters)

* In the file [`great_expectations/render/view/view.py`](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/great_expectations/render/view/view.py):

> [Line 145:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/great_expectations/render/view/view.py#L145): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (126 > 120 characters)

* In the file [`tests/cli/test_datasource_pandas.py`](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/cli/test_datasource_pandas.py):

> [Line 68:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/cli/test_datasource_pandas.py#L68): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (136 > 120 characters)

* In the file [`tests/cli/test_datasource_sqlite.py`](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/cli/test_datasource_sqlite.py):

> [Line 52:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/cli/test_datasource_sqlite.py#L52): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (130 > 120 characters)













* In the file [`tests/integration/usage_statistics/instantiate_context_with_usage_statistics.py`](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/integration/usage_statistics/instantiate_context_with_usage_statistics.py):

> [Line 57:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/integration/usage_statistics/instantiate_context_with_usage_statistics.py#L57): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (143 > 120 characters)

* In the file [`tests/integration/usage_statistics/test_usage_statistics.py`](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/integration/usage_statistics/test_usage_statistics.py):

> [Line 113:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/integration/usage_statistics/test_usage_statistics.py#L113): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (134 > 120 characters)
> [Line 155:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/integration/usage_statistics/test_usage_statistics.py#L155): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (134 > 120 characters)

* In the file [`tests/render/test_data_documentation_site_builder.py`](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/render/test_data_documentation_site_builder.py):

> [Line 220:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/render/test_data_documentation_site_builder.py#L220): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (127 > 120 characters)
> [Line 237:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/render/test_data_documentation_site_builder.py#L237): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (152 > 120 characters)
> [Line 240:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/render/test_data_documentation_site_builder.py#L240): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (133 > 120 characters)
> [Line 245:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/render/test_data_documentation_site_builder.py#L245): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (177 > 120 characters)
> [Line 262:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/render/test_data_documentation_site_builder.py#L262): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (127 > 120 characters)
> [Line 279:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/render/test_data_documentation_site_builder.py#L279): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (152 > 120 characters)
> [Line 282:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/render/test_data_documentation_site_builder.py#L282): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (133 > 120 characters)
> [Line 287:121](https://github.com/great-expectations/great_expectations/blob/0a4b7ca6f3d907c45bce5c0b6612f1798852ae4d/tests/render/test_data_documentation_site_builder.py#L287): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (126 > 120 characters)

This linter comment has been updated based on the most recent commit.

##### Comment last updated at 2020-03-17 08:06:42 UTC

roblim committed:
Set level on module logger instead

roblim committed:
standardize around “usage_statistics”

- change “telemetry” to “usage_statistics”
- change “anonymized_usage_data” to “anonymized_usage_statistics”

roblim committed:
Return DataContextConfig obj instead of dict

- update usages

roblim committed:
Add anonymized_usage_statistics property

roblim committed:
Update tests

roblim committed:
Compare anonymized_usage_statistics attributes instead of objects

roblim committed:
Change back to critical
